### PR TITLE
Cleanup MenuItem and add param names

### PR DESCRIPTION
**Pull request by hughbe**
All this is already tested

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/MenuItem.cs [hunk 1]
```diff
@@ -1242,347 +1164,332 @@ private struct MsaaMenuInfoWithId {
...
-        protected virtual void OnInitMenuPopup(EventArgs e) {
-            OnPopup(e);
-        }
+        protected virtual void OnInitMenuPopup(EventArgs e) => OnPopup(e);
 
-        // C#r
-        internal virtual void _OnInitMenuPopup( EventArgs e ) {
-            OnInitMenuPopup( e );
-        }
+        internal void OnInitMenuPopupInternal(EventArgs e) => OnInitMenuPopup(e);
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/MenuItem.cs]: An internal calls a protected calls another protected? Why is this internal necessary? make the protected a protected internal

File: src/System.Windows.Forms/src/System/Windows/Forms/MenuItem.cs [hunk 2]
```diff
@@ -2,907 +2,777 @@
...
+                return (_data.State & StateDefault) != 0;
             }
-            set {
+            set
+            {
                 CheckIfDisposed();
-                if (menu != null) {
-                    if (value) {
-                        UnsafeNativeMethods.SetMenuDefaultItem(new HandleRef(menu, menu.handle), MenuID, false);
+                if (_menu != null)
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/MenuItem.cs]: Should use Menu b/c _menu is private and this method is public

Comment (zsd4yr) [reply src/System.Windows.Forms/src/System/Windows/Forms/MenuItem.cs]: This is the case with many instances of using _menu in an inappropriate context

File: src/System.Windows.Forms/src/System/Windows/Forms/Menu.cs [hunk 3]
```diff
@@ -300,7 +300,7 @@ public abstract class Menu : Component {
                         item.Site.Container.Remove(item);
                     }
 
-                    item.Menu = null;
+                    item.Parent = null;
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/Menu.cs]: why was this changed?

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/Menu.cs]: Previously we defined `Menu` as:
```
        internal Menu Menu
        {
            get => _menu;
            set
            {
                _menu = value;
#if DEBUG
                _debugParentMenu = value;
#endif
            }
        }
```

This is equivalent to `Parent`, which is defined as
```
        public Menu Parent { get; internal set; }
```

**Comment by zsd4yr** [comment]
@hughbe please build locally before pushing.
